### PR TITLE
[stats] Rewrite stats to use MySQL

### LINKED ISSUE #16: [stats] Send day stats before clearing
Once `stats` is merged, add a send message to `clear()`. 

i.e. right before the bot clears daily stats, it sends the day's stats.

### PR DESCRIPTION
**Pull request by obviyus**
stats will now use MySQL to maintain a database of users vs. message count. Additionally, at `clear` bot will send the day's stats to the chats it belongs to. Fixes #16.

### CODE REVIEW

File: api/stats.py [hunk 1]
```diff
@@ -1,79 +1,85 @@
...
-    if user_object not in stats_dict[chat_id]:
-        stats_dict[chat_id][user_object] = 1
+    update.message.reply_text(text=text)
 
-    else:
-        stats_dict[chat_id][user_object] += 1
 
+def clear(update):
+    """"Reset message count to 0 for a chat"""
+    print_stats(update, None)
```
Comment (Tyagdit) [inline api/stats.py]: I don't think everyone would like this

File: api/stats.py [hunk 2]
```diff
@@ -1,79 +1,85 @@
...
+        """)
+    return cursor.fetchone()[0] == 1
 
-def stats_check(update, context):
-    global stats_dict
-    # from global_stats import global_stats_dict
 
-    msg = update.message
+def print_stats(update, context):
+    """Get daily chat stats"""
```
Comment (Tyagdit) [inline api/stats.py]: We should specify that this is in IST

File: api/stats.py [hunk 3]
```diff
@@ -1,79 +1,85 @@
...
+        text = "No messages found."
 
-    if user_object not in stats_dict[chat_id]:
-        stats_dict[chat_id][user_object] = 1
+    update.message.reply_text(text=text)
 
-    else:
-        stats_dict[chat_id][user_object] += 1
 
+def clear(update):
```
Comment (Tyagdit) [inline api/stats.py]:
```suggestion
def clear(context):
```

File: api/stats.py (hunk 1 above)
Comment (Tyagdit) [reply api/stats.py]: oh this also wouldn't work since there's no update

**Review by Tyagdit — CHANGES_REQUESTED** [review]
Haven't reviewed thoroughly but why is spurdo gone?

**Comment by obviyus** [comment]
> Haven't reviewed thoroughly but why is spurdo gone?

The API we were using has been taken down (404). We could try to implement our own logic for it.

**Review by Isocyanide — APPROVED** [review]

**Review by Tyagdit — APPROVED** [review]

**Pull request merged**